src/daft-core/src/array/ops/utf8.rs

```diff
@@ -827,25 +838,33 @@ impl Utf8Array {
             ));
         }
 
-        fn rpad_str(val: &str, length: usize, fillchar: &str) -> DaftResult<String> {
+        fn pad_str(
+            val: &str,
+            length: usize,
+            fillchar: &str,
+            placement: PadPlacement,
+        ) -> DaftResult<String> {
             if val.chars().count() >= length {
                 return Ok(val.chars().take(length).collect());
             }
             let fillchar = if fillchar.is_empty() {
                 return Err(DaftError::ComputeError(
-                    "Error in rpad: empty pad character".to_string(),
+                    "Error in pad: empty pad character".to_string(),
                 ));
             } else if fillchar.chars().count() > 1 {
                 return Err(DaftError::ComputeError(format!(
-                    "Error in rpad: {} is not a valid pad character",
+                    "Error in pad: {} is not a valid pad character",
                     fillchar.len()
                 )));
             } else {
                 fillchar.chars().next().unwrap()
             };
             let fillchar =
                 std::iter::repeat(fillchar).take(length.saturating_sub(val.chars().count()));
-            Ok(val.chars().chain(fillchar).collect())
+            Ok(match placement {
+                PadPlacement::Left => fillchar.chain(val.chars()).collect(),
+                PadPlacement::Right => val.chars().chain(fillchar).collect(),
+            })
```

Comment:
Small nit: I would recommend trying to do this match outside of the iteration so that we don't have an extra branching statement in the hot path

Reply:
Updated.